src/Microsoft.Health.Fhir.Shared.Client/IFhirClient.cs

```diff
@@ -39,7 +39,14 @@ Task<FhirResponse> DeleteAsync<T>(T resource, CancellationToken cancellationToke
         Task<FhirResponse> HardDeleteAsync<T>(T resource, CancellationToken cancellationToken = default)
             where T : Resource;
 
-        Task<FhirResponse> PatchAsync(string uri, string content, string ifMatchVersion = null, CancellationToken cancellationToken = default);
+        Task<FhirResponse<T>> PatchAsync<T>(T resource, string content, string ifMatchVersion = null, CancellationToken cancellationToken = default)
+            where T : Resource;
+
+        Task<FhirResponse<T>> PatchAsync<T>(string uri, string content, string ifMatchVersion = null, CancellationToken cancellationToken = default)
```

Comment:
Do we use this from the interface anywhere?

Reply:
Not really.

I added it just to copy 
` Task<FhirResponse<T>> CreateAsync<T>(string uri,`
` Task<FhirResponse<T>> UpdateAsync<T>(string uri,`